## Dataset Overview

In [1]:
import pandas as pd

df = pd.read_excel("../data/ApexPlanet_DataAnalytics_Dataset.xlsx")
df.shape

(1000, 12)

In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Order_ID       1000 non-null   str    
 1   Order_Date     1000 non-null   str    
 2   Customer_ID    1000 non-null   str    
 3   Customer_Name  1000 non-null   str    
 4   Age            980 non-null    float64
 5   Gender         1000 non-null   str    
 6   City           987 non-null    str    
 7   Product        1000 non-null   str    
 8   Category       1000 non-null   str    
 9   Quantity       1000 non-null   int64  
 10  Unit_Price     1000 non-null   float64
 11  Total_Sales    1000 non-null   float64
dtypes: float64(3), int64(1), str(8)
memory usage: 93.9 KB


### Observation

The dataset contains 1000 records and 12 variables related to retail sales transactions.

The dataset includes customer demographics, product details, and transaction information. Most columns are categorical, while Age, Quantity, Unit_Price, and Total_Sales are numerical variables.

The Order_Date column is currently stored as a string and will require conversion to datetime format for time-based analysis.

## Descriptive Statistics

In [8]:
df.describe()

,Age,Quantity,Unit_Price,Total_Sales
count,980.000000,1000.000000,1000.000000,1000.000000
mean,41.360204,5.435000,25486.783410,139399.439650
std,13.822597,2.838632,14179.402361,114100.051546
min,18.000000,1.000000,145.780000,437.340000
25%,30.000000,3.000000,13895.722500,47066.632500
50%,41.000000,5.000000,25398.740000,108594.025000
75%,54.000000,8.000000,37512.382500,203722.882500
max,65.000000,10.000000,49997.530000,493677.500000


### Observation

The average customer age is approximately 41 years, with ages ranging from 18 to 65 years.

Customers purchase around 5 to 6 units per transaction on average.

The mean Total_Sales value (₹139,399) is higher than the median value (₹108,594), suggesting the presence of high-value transactions that may influence the overall sales distribution.

Further exploratory analysis will be performed to identify sales patterns and potential outliers.

## Data Quality Assessment

In [4]:
df.isnull().sum()

Order_ID          0
Order_Date        0
Customer_ID       0
Customer_Name     0
Age              20
Gender            0
City             13
Product           0
Category          0
Quantity          0
Unit_Price        0
Total_Sales       0
dtype: int64

In [5]:
df.duplicated().sum()

np.int64(0)

### Observation

Missing values were detected in the Age (20 records) and City (13 records) columns.

No duplicate records were found in the dataset, indicating good data consistency.

The amount of missing data is relatively small and can be handled during the data preparation stage.